### ISSUE TITLE
How to calculate model.predict accuracy?

### ISSUE THREAD

**Issue by SAyaka0122** [4uiiurz1/keras-arcface#5]
Hi,
Thanks for sharing the great code.
I would like to calculate the accuracy score of model.predict as follow but I don't think it work well.
Could you please let me know how to solve this problem??

```
import time
import os
import math
import argparse
from glob import glob
from collections import OrderedDict
import random
import warnings
from datetime import datetime
import joblib
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D


import numpy as np
import pandas as pd

from sklearn.model_selection import KFold, StratifiedKFold
from sklearn.manifold import TSNE
from sklearn.decomposition import PCA
from sklearn.metrics import accuracy_score
from sklearn import metrics
from sklearn.metrics import confusion_matrix

import keras
from keras.datasets import mnist
from keras.preprocessing.image import ImageDataGenerator
from keras.models import Sequential, load_model, Model
from keras.layers import Dense, Dropout, Activation, Flatten
from keras.layers import Conv2D, MaxPooling2D
from keras.models import load_model
from keras.models import model_from_yaml
from keras.optimizers import SGD, Adam
from keras.callbacks import EarlyStopping, ModelCheckpoint, ReduceLROnPlateau, CSVLogger, LearningRateScheduler
from keras.utils import np_utils

import archs
from metrics import *
from scheduler import *

import h5py

(X, y), (X_test, y_test) = mnist.load_data()

X = X[:, :, :, np.newaxis].astype('float32') / 255
X_test = X_test[:, :, :, np.newaxis].astype('float32') / 255
print('X:', X.shape) # X: (60000, 28, 28, 1)
print('X_test.shape:', X_test.shape)# X_test.shape: (10000, 28, 28, 1)
print('X_test.ndim:', X_test.ndim)# X_test.ndim: 4
y_ohe = keras.utils.to_categorical(y, 10)
print('y_ohe:', y_ohe.shape)# y_ohe: (60000, 10)
y_ohe_test = keras.utils.to_categorical(y_test, 10)
print('y_ohe_test.shape:', y_ohe_test.shape)# y_ohe_test.shape: (10000, 10)
print('y_ohe_test.ndim:', y_ohe_test.ndim)# y_ohe_test.ndim: 2

arcface_model = load_model('./models/mnist_vgg8_arcface_3d/model.hdf5', custom_objects={'ArcFace': ArcFace})
arcface_model = Model(inputs=arcface_model.input[0], outputs=arcface_model.layers[-3].output)
print('arcface_model.inputs:', arcface_model.inputs)# arcface_model.inputs: [<tf.Tensor 'input_1_11:0' shape=(?, 28, 28, 1) dtype=float32>]
print('arcface_model.outputs:', arcface_model.outputs)# arcface_model.outputs: [<tf.Tensor 'batch_normalization_8_11/cond/Merge:0' shape=(?, 3) dtype=float32>]
arcface_features = arcface_model.predict(X_test, verbose=1)
arcface_features /= np.linalg.norm(arcface_features, axis=1, keepdims=True)
print('arcface_features.shape:', arcface_features.shape)# arcface_features.shape: (10000, 3)

y_pred=np.argmax(arcface_features, axis=1)
print('y_pred.shape:', y_pred.shape)# y_pred.shape: (10000,)
print('y_pred:', y_pred)# y_pred: [0 2 1 ... 0 2 2] # I think this is the cause of problem because there are only 3 dims against 10 classes.
true_classes = np.argmax(y_ohe_test, axis=1)
print('true_classes.shape:', true_classes.shape)# true_classes.shape: (10000,)
print('true_classes:', true_classes)# true_classes: [7 2 1 ... 4 5 6]

# display the accuracy
accuracy = accuracy_score(true_classes, y_pred)
print('{:.2%}'.format(accuracy))# 22.85%
```

Thank you!

**Comment by farhantandia** [comment]
hey, how you solve this problem? thanks
